src/librustdoc/html/render/mod.rs

```diff
@@ -3631,6 +3636,12 @@ fn render_deref_methods(
     let what =
         AssocItemRender::DerefFor { trait_: deref_type, type_: real_target, deref_mut_: deref_mut };
     if let Some(did) = target.def_id() {
+        if let Some(type_did) = impl_.inner_impl().for_.def_id() {
+            if did == type_did {
+                // Avoid infinite cycles
```

Comment:
Oh wow, it's horrifying that this is legal :laughing: Excellent catch.

Can you add this as a test case? https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d4a7c811ce01c78d711c495a67473b01

Reply:
Haha, yeah, I wouldn't have thought it was either... 😅 I only happened to discover it because of an [existing test](https://github.com/rust-lang/rust/blob/master/src/test/rustdoc/search-index.rs) that seems to (accidentally?) make use of this. Anyway, I'll add your test as well.

Reply:
Oh, if there's an existing test than this is fine; maybe test the generated docs look reasonable but otherwise there's no sense testing it twice.

Reply:
Eh, I'll add yours too. I think it makes things a bit more obvious that way, esp. if the existing test is ever changed so that it no longer triggers this case. 😄